plugins/registry.js

```diff
@@ -79,6 +79,29 @@ export default class PluginRegistry {
         return dispatchPluginComponentAction('LeftSidebarHeader', this.id, component);
     }
 
+    // Register an App in the list of apps.
+    // Accepts a React component. Returns a unique identifier.
+    registerTeamAppComponent(component, show = () => true) {
+        const id = generateId();
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'TeamApp',
+            data: {
+                id,
+                pluginId: this.id,
+                component,
+                show,
+            },
+        });
+        return id;
+    }
+
+    // Register App visualization in the center panel.
+    // Accepts an id for the url and a React component. Returns a unique identifier.
+    registerAppCenterComponent(id, component) {
```

Comment:
Yes, it is. The idea is that you can quickly access plugins both from the sidebar menu and the channel header plugin dropdown, making them show or not depending on the channel. Then these plugins can take full control of the whole chat area to show any type of functionality if they need to. We have a full wiki per channel and the kanban functionality, among others.

So it is an application functionality contained in a plugin. What can we call that?

Reply:
Slack uses the term 'spaces', but I'm open to discussing further.

Reply:
so, it would be `SpaceComponent` and `registerSpaceComponent`?  Not bad, even though I can imagine that anyone using this hook will be doing so to add some application-like functionality to the center space I still think space seems good.

What about the side bar menu? `registerTeamSpaceComponent` ?